### DISCUSSION TITLE
Where generated CSS is stored

### DISCUSSION THREAD

**Discussion by ghost** [master-co/css#36] [Question]
I just wonder where all the CSS generated by JIT is stored? For example, if I add `w:full` to the element, it will generate `width: 100%` as I inspect it by the inspector of the browser. It points to the file `master.css`, but as I click the file and search `width: 100%`, there is actually no such rule.

**Comment by 1aron** [comment]
Take our biggest feature JIT for example, it scans the entire DOM Tree at runtime and automatically generates CSS Rules in `<style id="master-css">` according to our rules engine.

![Screen Shot 2022-07-02 at 12 07 06 AM](https://user-images.githubusercontent.com/33840671/176930939-49b3c766-5826-41c0-88cd-287a93a635df.jpg)

This rules engine is completely created by us, so if you're curious how it works, the most direct way is to look at our core code.

**Reply by ghost** [reply]
Why I cannot find `<style id="master-css">...</style>` by the inspector and page source...I only found one file called master.css but inside there are no CSS rules that should be generated...Instead looks like some reset CSS.

**Reply by 1aron** [reply]
It's very strange since we didn't generate any file called `master.css` and probably need your screenshot.

**Comment by ghost** [comment]
I'm not using the CDN but I download the files and use it locally, not sure if this is the reason?
For example, if I inspect the element, as you can see that the generated CSS is in the file called `master.css`

![screenshot-1](https://user-images.githubusercontent.com/73235650/177005773-fb701540-1b69-4f19-83ca-53b9548c7f96.png)

When I click that file, there are actually all about resetting the CSS, but doesn't include any CSS rule that should be generated by the class name given, which is why I wonder and ask where are all the CSS generated are stored.

![screenshot-2](https://user-images.githubusercontent.com/73235650/177005830-99c45310-b966-47ce-9d2f-753db676b8fa.png)

**Reply by 1aron** [reply]
Check out the latest documentation [Runtime Rendering](https://beta.css.master.co/docs/rendering-modes/runtime-rendering).
